### PR TITLE
[Page Actions] Fix render issues

### LINKED ISSUE #7330: [Page] Title and secondary actions flicker on form state change
## Issue summary
On pages with long title and collapsed secondary actions, the page flickers when its state changes

### Examples

**Product details**

https://user-images.githubusercontent.com/15054990/193677624-546c7479-2cc8-4868-9591-6d43eda83928.mp4

**Discount details**

https://user-images.githubusercontent.com/15054990/193677638-69bfb480-6459-42b3-bc3b-1c2dbce6cfdb.mp4

**Product variant details**

https://user-images.githubusercontent.com/15054990/193682665-6fdf243b-dd2b-482a-9eeb-3898ebf19633.mp4

## Expected behaviour
Page doesn't flicker when its state changes 

## Actual behaviour
Page flickers when its state changes 

## Steps to reproduce the problem
1. Open an admin page that renders `<Page />` component and has multiple secondary actions (e.g. product details page, discount details page and others)
1.1. Page should have secondary actions collapsed into More actions dropdown
1.2. Having a long title makes the issue more prominent, however, it's not a pre-req for reproducing the issue, it's the secondary actions dropdown itself that flickers
2. Change the page state (e.g. form field value)
3. See the page actions flicker and re-render the popover that's used in the collapsed secondary actions

### PR DESCRIPTION
**Pull request by kyledurand**
Fixes: https://github.com/Shopify/polaris/issues/7330
[Spin link](https://admin.web.web-mg6g.kyle-durand.us.spin.dev/store/shop1/)

### CODE REVIEW

File: polaris-react/src/components/ActionMenu/components/Actions/Actions.tsx [hunk 1]
```diff
@@ -169,12 +175,13 @@ export function Actions({actions = [], groups = [], onActionRollup}: Props) {
     [measureActions],
   );
 
-  useEffect(() => {
-    if (!actionsLayoutRef.current) {
-      return;
-    }
+  useEventListener('resize', handleResize);
+
+  useLayoutEffect(() => {
```
Comment (kyledurand) [inline polaris-react/src/components/ActionMenu/components/Actions/Actions.tsx:180]: This part of the PR is the fix. The rest is just cleanup and changing our details page to better mimic product details

[From docs](https://reactjs.org/docs/hooks-reference.html#uselayouteffect):
> The signature is identical to useEffect, but it fires synchronously after all DOM mutations. Use this to read layout from the DOM and synchronously re-render. Updates scheduled inside useLayoutEffect will be flushed synchronously, before the browser has a chance to paint.

One of the suggestions from @kushp was to use [flushSync](https://beta.reactjs.org/apis/react-dom/flushSync) to defer the state update but isn't great for performance. `useLayoutEffect` seems to do the same thing but from what I've read in a different way that shouldn't affect performance. This waits until DOM stuff has finished before painting, removing layout thrashing

**Review by henryyi — APPROVED** [review]
🎩 and changes look good!

**Review by tmlayton — APPROVED** [review]

**Pull request merged**